infrastructure/src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -118,12 +121,21 @@
      *
      * {@link SequencerServer::maxConflictWildcard} :
      * a "wildcard" representing the maximal update timestamp of
-     * all the confict keys which were evicted from the cache
+     * all the conflict keys which were evicted from the cache
      */
     private long maxConflictWildcard = Address.NOT_FOUND;
 
+    /**
+     * Represent the max update timestamp of all the conflict keys
+     * which were evicted from the cache by the time this server was elected
+     * the primary sequencer. This means that any snapshot timestamp below this
+     * actual threshold would abort due to NEW_SEQUENCER cause.
+     */
+    private long maxConflictNewSequencer = Address.NOT_FOUND;
```

Comment:
How are you detecting this, are you just marking the initial token (i.e. during initialization) and throwing new sequencer whenever you see a timestamp less than that? I don't think thats unique enough to identify this issue. Can we use the sequencer epoch?

Reply:
A timestamp is tied to a sequencer epoch, if the epoch != the current sequencer's epoch then reject as a different sequencer. In this case, the epoch also marks the sequencers in memory state.

Reply:
I am detecting this upon a full bootstrap sequencer message (resetServer method) which implies that this is a newly elected Sequencer. As soon as we determine that this is a newly elected sequencer we set this boundary, to the same limit of the last evicted entry in the cache meaning that we are not able to resolve anything below this point but the actual reason is because it is a NEW_SEQUENCER and not actually because OVERFLOW. As soon as you start removing things from the cache the maxConflictWildcard distances away from this boundary and may lead to aborts due to overflow.